sdk/python/feast/feature_store.py

```diff
@@ -1569,7 +1581,7 @@ def _get_online_features(
         }
 
         # If the embedded Go code is enabled, send request to it instead of going through regular Python logic.
-        if self.config.go_feature_retrieval:
+        if self.config.go_feature_retrieval and self._go_server:
```

Comment:
@achals What was the reason for adding `and self._go_server` to this condition expression?  This seems to be the cause of why the go server will not start and trying to understand if there was another issue that was being solved for.